prog/rand.go

```diff
@@ -87,14 +95,30 @@ func (r *randGen) randInt() uint64 {
 	case r.nOutOf(5, 7):
 		v = uint64(-int64(v))
 	default:
-		v <<= uint(r.Intn(63))
+		bits := int(8 * size)
+		v <<= uint(r.Intn(bits - 1))
 	}
-	return v
+	return v & limit
+}
+
+func getSizeLimit(size uint64) uint64 {
```

Comment:
`Mask` is a more common name for a value that is used with `&` operation to mask low/high bits.
But I think doing something like `func truncateToSize(v, size uint64) uint64)` will be better as it will encapsulate the `&` operation too.